src/broker/Manager.cc

```diff
@@ -813,14 +813,32 @@ bool Manager::AutoUnpublishEvent(const string& topic, Val* event) {
 }
 
 RecordVal* Manager::MakeEvent(ValPList* args, zeek::detail::Frame* frame) {
-    auto rval = new RecordVal(BifType::Record::Broker::Event);
+    // Deprecated MakeEvent() version using ValPList - requires extra copy.
+    zeek::Args cargs;
+    cargs.reserve(args->size());
+    for ( auto* a : *args )
+        cargs.push_back({zeek::NewRef{}, a});
+
+    return MakeEvent(cargs.begin(), cargs.end(), frame)->Ref()->AsRecordVal();
+}
+
+zeek::RecordValPtr Manager::MakeEvent(ArgsIt first, ArgsIt last, zeek::detail::Frame* frame) {
+    scoped_reporter_location srl{frame};
+    return MakeEvent(first, last);
+}
+
+zeek::RecordValPtr Manager::MakeEvent(ArgsIt first, ArgsIt last) {
+    auto rval = zeek::make_intrusive<RecordVal>(BifType::Record::Broker::Event);
     auto arg_vec = make_intrusive<VectorVal>(vector_of_data_type);
     rval->Assign(1, arg_vec);
     Func* func = nullptr;
-    scoped_reporter_location srl{frame};
 
-    for ( auto i = 0; i < args->length(); ++i ) {
-        auto arg_val = (*args)[i];
+    auto it = first;
+
+    int argc = std::distance(first, last);
```

Comment:
This is a narrowing bug on 64-bit platforms (`distance` returns `ptrdiff_t`, not `int`). Simply use `auto` instead:

```suggestion
    auto argc = std::distance(first, last);
```